optuna/storages/grpc/_client.py

```diff
@@ -0,0 +1,314 @@
+from __future__ import annotations
+
+from collections.abc import Container
+from collections.abc import Iterable
+from collections.abc import Sequence
+import json
+from typing import Any
+import uuid
+
+from optuna.distributions import BaseDistribution
+from optuna.distributions import distribution_to_json
+from optuna.exceptions import DuplicatedStudyError
+from optuna.storages._base import BaseStorage
+from optuna.storages._base import DEFAULT_STUDY_NAME_PREFIX
+from optuna.storages.grpc._grpc_imports import _imports
+from optuna.storages.grpc._server import _from_proto_trial
+from optuna.storages.grpc._server import _to_proto_trial
+from optuna.storages.grpc._server import _to_proto_trial_state
+from optuna.study._frozen import FrozenStudy
+from optuna.study._study_direction import StudyDirection
+from optuna.trial._frozen import FrozenTrial
+from optuna.trial._state import TrialState
+
+
+if _imports.is_successful():
+    from optuna.storages.grpc._grpc_imports import api_pb2
+    from optuna.storages.grpc._grpc_imports import api_pb2_grpc
+    from optuna.storages.grpc._grpc_imports import grpc
+
+
+def _handle_grpc_error(e: grpc.RpcError) -> None:
+    if e.code() == grpc.StatusCode.NOT_FOUND:
+        raise KeyError from e
+    elif e.code() == grpc.StatusCode.ALREADY_EXISTS:
+        raise DuplicatedStudyError from e
+    elif e.code() == grpc.StatusCode.FAILED_PRECONDITION:
+        raise RuntimeError from e
+    elif e.code() == grpc.StatusCode.INVALID_ARGUMENT:
+        raise ValueError from e
+    else:
+        raise e
```

Comment:
@HideakiImamura @porink0424  am concerned about introducing this function to handle gRPC errors because the mapping between gRPC status codes and exceptions can vary depending on the context. For example, `grpc.StatusCode.ALREADY_EXISTS` is currently mapped to `optuna.exceeptions.DuplicatedStudyError`, but this logic is very specific to the current specification and might break if the specification changes in the future. What do you think?

Reply:
Thanks for the comment. I will change this error handling as before.